client/src/components/GraphViewerComponent/GraphViewerComponent.js

```diff
@@ -281,7 +281,10 @@ class GraphViewerComponent extends React.Component {
 
       const twinsChunks = this.modelService.chunkModelsList(currentTwins, 100);
       const bs = new BatchService({
-        refresh: () => this.cyRef.current.doLayout(),
+        refresh: () => {
+          this.cyRef.current.doLayout();
+          this.cyRef.current.setNewNodesInitialPositions();
```

Comment:
`refresh` can be called multiple times during layout. It might be better to call this function after `await bs.run()` below? Unless there is a reason to call it multiple times?

Reply:
Done, Changed and tested.